qi-lib/private/util.rkt

```diff
@@ -137,6 +138,31 @@
 (define (filter-values f . args)
   (apply values (filter f args)))
 
+(define (partition-values c+bs . args)
```

Comment:
nit: Would be good to add some comments to explain how this function works at a high level

Reply:
Erm, yes. It was written in a fugue of "flow state" ;) and never got the attention it deserved.